src/commands.ts

```diff
@@ -60,51 +62,58 @@ async function buildCurrentProject(args, dotNetSdk: DotNetInfo, commandObserver:
+        commandObserver.buildInProgress = false;
+    });
+}
+
+async function buildAllProjects(dotNetSdk: DotNetInfo, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken): Promise<void> {
+    let projects = await vscode.workspace.findFiles('{**/*.pgproj}');
+    await buildProjects(dotNetSdk, projects.map(p => p.fsPath), commandObserver, cancelToken);
+}
+
+async function buildProjects(dotNetSdk: DotNetInfo, projects: string[], commandObserver: CommandObserver, cancelToken: vscode.CancellationToken) {
     commandObserver.clear();
+    let unsupportedProjects = await validateProjectSDK(projects, commandObserver);
+    if (unsupportedProjects.length > 0) {
+        projects = projects.filter(p => unsupportedProjects.indexOf(p) < 0);
+    }
 
-    let isValid = await validateProjectSDK([project], commandObserver);
-    if (!isValid) {
+    if (commandObserver.buildInProgress) {
+        vscode.window.showErrorMessage(Constants.existingBuildInProgressMessage);
         return;
     }
-    await dotnetBuild(dotNetSdk, project, commandObserver, cancelToken);
-}
 
-async function buildAllProjects(dotNetSdk: DotNetInfo, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken): Promise<void> {
     try {
-        let projects = await vscode.workspace.findFiles('{**/*.pgproj}');
-        commandObserver.clear();
-
-        let isValid = await validateProjectSDK(projects.map(p => p.fsPath), commandObserver);
-        if (!isValid) {
-            return;
-        }
-
+        commandObserver.buildInProgress = true;
         for (let project of projects) {
             if (cancelToken.isCancellationRequested) {
                 return;
             }
-            await dotnetBuild(dotNetSdk, project.fsPath, commandObserver, cancelToken);
+            await dotnetBuild(dotNetSdk, project, commandObserver, cancelToken);
         }
+        commandObserver.buildInProgress = false;
     } catch (err) {
         vscode.window.showErrorMessage(err);
+        commandObserver.buildInProgress = false;
     }
 }
 
-async function validateProjectSDK(projects: string[], commandObserver: CommandObserver): Promise<boolean> {
+async function validateProjectSDK(projects: string[], commandObserver: CommandObserver): Promise<string[]> {
     var packageInfo = Utils.getPackageInfo();
     let unsupportedProjects = await Helper.checkProjectVersion(packageInfo.minSupportedPostgreSQLProjectSDK, packageInfo.maxSupportedPostgreSQLProjectSDK, projects, commandObserver);
     if (unsupportedProjects && unsupportedProjects.length > 0) {
-        unsupportedProjects.map(p => commandObserver.logToOutputChannel(strings.format(Constants.buildFailedUnsupportedSdkMessage, p)));
-        return Promise.resolve(false);
+        unsupportedProjects.map(p =>
+            commandObserver.logToOutputChannel(localize('extension.projectUpdateFailedMessage', 'Failed to build project {0}.\nUpdate PostgreSQL SDK to latest version.\n', p)));
```

Comment:
Can we give them range of supported versions here?

I think "update to latest" might be miss-guiding, if they try to open newer projects with older extension. Would be nice if we can give them a generic blurb, like ```Cannot build project with SDK version X. Installed PostgreSQL extension only supports SDK versions between Y and Z.```
But for that we will need to:
1) return project version from the validation method
2) get access to the supported version range from here
so I'm not sure how easy it is...